document_merge_service/api/engines.py

```diff
@@ -19,11 +19,28 @@ def validate(self):
         self.template.seek(0)
 
 
+def walk_nested(data, fn):
+    """Apply fn to every primitive value in nested data."""
+    if isinstance(data, dict):
+        return {key: walk_nested(value, fn) for (key, value) in data.items()}
+    if isinstance(data, list):
+        return [walk_nested(value, fn) for value in data]
+    return fn(data)
+
+
 class DocxTemplateEngine(DocxValidator):
     def __init__(self, template):
         self.template = template
 
     def merge(self, data, buf):
+        def wrap(value):
```

Comment:
I'd name this `transform` since you just take a value and transform it to a listing